### PR TITLE
add github action

### PR DESCRIPTION
**Pull request by hayguen**
* build on Ubuntu Linux
* cross build from Ubuntu for win32/win64 targets
* get MINGW cross compile run - setting target architecture
* build on Windows with MSVC (Visual Studio)

Signed-off-by: hayati ayguen <h_ayguen@web.de>

### CODE REVIEW

File: .github/workflows/c-cpp.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      # Note the current convention is to use the -S and -B options here to specify source
+      # and build directories, but this is only available with CMake 3.13 and higher.
+      # The CMake binaries on the Github Actions machines are (as of this writing) 3.12
+      run: cmake $GITHUB_WORKSPACE -G "Visual Studio 16 2019" -A Win32 -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DUSE_SIMD=ON -DUSE_TYPE_FLOAT=ON -DUSE_TYPE_DOUBLE=ON -DUSE_BENCH_FFTW=OFF -DUSE_BENCH_GREEN=OFF -DUSE_BENCH_KISS=OFF -DUSE_BENCH_POCKET=OFF
+
+    - name: Build SIMD-full
+      working-directory: ${{runner.workspace}}/build_simd-full
+      shell: bash
+      # Execute the build.  You can specify a specific target with "--target <NAME>"
+      run: cmake --build . --config $BUILD_TYPE
```
Comment (marton78) [inline .github/workflows/c-cpp.yml:109]: You don't want to upload the build artefacts?

File: CMakeLists.txt [hunk 2]
```diff
@@ -136,18 +151,27 @@ if (USE_SIMD AND USE_SIMD_NEON)
...
+else()
+  if (USE_SIMD AND USE_TYPE_DOUBLE)
+    if(WIN32)
+      if(DISABLE_SIMD_AVX)
+        set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "/arch:SSE2")
+      else()
+        set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "/arch:AVX")
+      endif()
+    else()
+      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-march=native")
```
Comment (marton78) [inline CMakeLists.txt:170]: Why not `-march=core2`?

**Review by marton78 — APPROVED** [review]
Nice!

File: .github/workflows/c-cpp.yml (hunk 1 above)
Comment (hayguen) [reply .github/workflows/c-cpp.yml:109]: could be done and would be interesting - especially when providing binary builds on github releases.
but that would require some fiddling, how to remove all temorary files ..
however, my primary goal was testing compilation

File: CMakeLists.txt (hunk 2 above)
Comment (hayguen) [reply CMakeLists.txt:170]: that line with `-march=native` already existed, that i didn't think much about it.
could think it should depend on a cmake option like "optimize for local computer? (or minimum required, e.g. for distribution)". ideas on this?
core2: intrinsics for double could be using avx, which was '-march=sandybridge' as used for mingw cross compilation.
have the target a text/string cmake option?

Comment (marton78) [reply CMakeLists.txt:170]: I saw that you changed `-march=native` further above to `-march=core2`, that's why I asked. In general using `-march=native` is not a good idea for anyone who intends to package the build result -- your users might use a worse architecture than the machine you're building on.

It seems there is no built in option in CMake, as implied by [this SO question]( https://stackoverflow.com/questions/46724267/cmake-march-hardware).

**Comment by hayguen** [comment]
i'll check/go for a cmake option for setting target architecture with gcc .. (in the next days)

File: CMakeLists.txt (hunk 2 above)
Comment (unevens) [reply CMakeLists.txt:170]: in a [commit](https://github.com/dievsky/pffft/commit/d90577e0f260199a8f1f43835674a4628795443f) on the unaligned experiments branch by @dievsky, there is this:
    
    if(DISABLE_SIMD_AVX)
      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-msse2")
    else()
      set_property(SOURCE pffft_double.c PROPERTY COMPILE_FLAGS "-mavx")

i seem to remember that it was working with gcc on my debian machine, but i'm not sure now.

**Comment by hayguen** [comment]
going to replace this PR ..

**Pull request closed without merging**